api/v1alpha1/siteconfig_types.go

```diff
@@ -0,0 +1,461 @@
+	// after the installation is complete.
+	// +optional
+	NodeLabels map[string]string `json:"nodeLabels,omitempty"`
+
+	// Hostname is the desired hostname for the host
+	// +required
+	HostName string `json:"hostName,omitempty"`
+
+	//TODO: remove NodeIp
+	NodeIp string `json:"nodeIp,omitempty"`
+
+	// Provide guidance about how to choose the device for the image being provisioned.
+	// +kubebuilder:default:=UEFI
+	BootMode bmh_v1alpha1.BootMode `json:"bootMode,omitempty"`
+
+	// UserData holds the reference to the Secret containing the user data to be passed to the host before it boots.
+	// +optional
+	UserData corev1.SecretReference `json:"userData,omitempty"`
+
+	// Json formatted string containing the user overrides for the host's coreos installer args
+	// +optional
+	InstallerArgs string `json:"installerArgs,omitempty"`
+
+	// Json formatted string containing the user overrides for the host's ignition config
+	// IgnitionConfigOverride enables the assignment of partitions for persistent storage.
+	// Adjust disk ID and size to the specific hardware.
+	// +optional
+	IgnitionConfigOverride string `json:"ignitionConfigOverride,omitempty"`
+
+	// +kubebuilder:default:=master
+	Role string `json:"role,omitempty"`
+
+	// +optional
+	CrAnnotations CrAnnotations `json:"crAnnotations,omitempty"`
+
+	// +optional
+	DiskPartition []DiskPartition `json:"diskPartition,omitempty"`
+
+	// IronicInspect is used to specify if automatic introspection carried out during registration of BMH is enabled or disabled
+	// +kubebuilder:default:=""
+	// +optional
+	IronicInspect IronicInspect `json:"ironicInspect,omitempty"`
+
+	// +required
+	TemplateRefs []TemplateRef `json:"templateRefs,omitempty"`
+}
+
+const (
+	SNO      = "sno"
+	Standard = "standard"
+)
+
+// SiteConfigSpec defines the desired state of SiteConfig
+type SiteConfigSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// ClusterName is the name of the cluster.
+	// +required
+	ClusterName string `json:"clusterName,omitempty"`
+
+	// PullSecretRef is the reference to the secret to use when pulling images.
+	PullSecretRef *corev1.LocalObjectReference `json:"pullSecretRef,omitempty"`
+
+	ClusterImageSetNameRef string `json:"clusterImageSetNameRef,omitempty"`
+
+	// SSHPublicKey is the public Secure Shell (SSH) key to provide access to instances.
+	// This key will be added to the host to allow ssh access
+	// +optional
+	SshPublicKey string `json:"sshPublicKey,omitempty"`
+
+	// +optional
+	SshPrivateKeySecretRef SshPrivateKeySecretRef `json:"sshPrivateKeySecretRef,omitempty"`
+
+	// BaseDomain is the base domain to use for all clusters created in this pool.
+	// +required
+	BaseDomain string `json:"baseDomain,omitempty"`
+
+	// APIVIP is the virtual IP used to reach the OpenShift cluster's API.
+	// +optional
+	ApiVIP string `json:"apiVIP,omitempty"`
```

Comment:
I don't think this controller is ever going to coexist with a version of assisted or OCP so old that it won't have the plural. Maybe we just don't need to add the singular fields for these?

Reply:
I am okay with removing the singular fields, @imiller0 and @eranco74 - wdyt?

Reply:
I think, we only want to keep the plurals

Reply:
Sounds as though there is good support to remove the singular fields. I will remove them!